components/ContractViewer/DeploymentInfo.ts

```diff
@@ -0,0 +1,357 @@
+import { createContext, useContext, useState } from 'react'
+
+import solParser from '@solidity-parser/parser'
+import * as AstTypes from '@solidity-parser/parser/src/ast-types'
+// eslint-disable-next-line prettier/prettier
+import { type Abi } from 'abitype'
+import { EtherscanContractResponse, SolidityCompilerOutput } from 'types/contract'
+
+import { findContract, flattenCode } from 'util/flatten'
+import { solidityCompiler } from 'util/solc'
+
+import { SourceDefinition } from './AstProcessor'
+import EtherscanLoader from './EtherscanLoader'
+
+export type ParseResult = AstTypes.SourceUnit & {
+  errors?: any[]
+  tokens?: any[]
+}
+
+export type DeploymentsCollection = {
+  [address: string]: DeploymentInfo
+}
+
+// TODO: should probably move some of the contract code parts to another class
+// it might be useful to have separate compilationInfo for separate deployments but
+// this class is too big rn, can and probably should move code, abi, compilationInfo to
+// a "SolidityContract" class
+
+/**
+ * DeploymentInfo represents an on-chain deployment:
+ * 
+ * @property chainId (chain id as decimal number)

```

Comment:
Why suddenly start to use this format ? 🤔 

I don't have anything against it per se, but I feel like we should be consistent (we currently don't use this `@property` anywhere else)

Reply:
I copied it from somewhere because I thought I would get nice hints on vs code.
as you see I haven't ran with this format any longer since.

Reply:
hm okay - it's not that big of a deal, just a bit strange that we just have one place where we use this comment format.

Reply:
feel free to offer a standard and I will use it everywhere.